src/CashuWallet.ts

```diff
@@ -128,14 +165,15 @@ class CashuWallet {
 		}
 		return {
 			proofs,
-			tokensWithErrors: tokensWithErrors.length ? { token: tokensWithErrors } : undefined
+			tokensWithErrors: tokensWithErrors.length ? { token: tokensWithErrors } : undefined,
+			newKeys
```

Comment:
this function has become so weird. If it does receive tokens from multiple mints, it will only return the token from the 'active' mint. I'm not complaining about your implementation, i think it is good, but maybe it warrants the question if the protocol should support tokens from multiple mints inside a single request, or if that is too complicated. I guess it's functionality vs simplicity. 

Would love your thoughts on that @callebtc

Reply:
It's a good question. I think receiving from multiple mints should be pretty straightforward and actually one of the reasons why we ended up with V3 tokens.

As you can see [here in NUT-00](https://github.com/cashubtc/nuts/blob/main/00.md#example-json-2), we assume that each entry of `token.token` is from one mint, that means you should be able to loop through entries `token.token` and receive nuts. In pseudo-python:

```python
for t in token.token:
    activateMint(t.mint)
    redeem(t.proofs)
```

Does that make sense?

Reply:
yes this does make sense. @BilligsterUser maybe we have to refactor the receive function to only take proofs and a mint as an argument? and then let the wallets/implementations handle the complexity?